src/Orchard.Web/Modules/Orchard.OpenId/InMemoryCache.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+namespace Orchard.OpenId {
+    public class InMemoryCache : TokenCache {
```

Comment:
@sebastienros, I am having trouble understanding exactly how to convert this class to a Singleton service, and I couldn't find the documentation of this class, can you please provide me a link.